beacon-chain/core/light-client/store.go

```diff
@@ -24,13 +24,20 @@ type Store struct {
 	lastOptimisticUpdate interfaces.LightClientOptimisticUpdate // tracks the best optimistic update seen so far
 	p2p                  p2p.Accessor
 	stateFeed            event.SubscriberSender
+	nonFinalityCache     *lightClientCache
 }
 
 func NewLightClientStore(db iface.HeadAccessDatabase, p p2p.Accessor, e event.SubscriberSender) *Store {
+	cp, err := db.FinalizedCheckpoint(context.Background())
```

Comment:
You should pass an existing context to `NewLightClientStore` so that everything is cleaned up properly. We only use `context.Backround()` in very specific scenarios, without a good reason you should pass down existing contexts